hook-common/src/webhook.rs

```diff
@@ -134,3 +134,22 @@ pub struct WebhookJobMetadata {
     pub plugin_id: Option<i32>,
     pub plugin_config_id: Option<i32>,
 }
+
+/// An error originating during a Webhook Job invocation.
+#[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
+pub struct WebhookJobError {
+    pub r#type: String,
+    pub details: String,
```

Comment:
`error_details` actually has an expected shape and fields that are used for things, if you want to copy from my Kafka message types.

Reply:
Interesting.

I was looking at the [ddl query](https://github.com/PostHog/posthog/blob/35cd4656b91d705215c9f3366e6a5509774318f6/posthog/models/app_metrics/sql.py#L15) for `app_metrics` when writing this. Looks like everything in details is dumped into a string in Clickhouse. In other words, plugin-server is the one ensuring the schema of `error_details` matches whatever is required, Clickhouse just stores and returns a string.